resources/js/Pages/BatchGrades/GradeFullscreenPage.vue

```diff
@@ -0,0 +1,154 @@
+import FullscreenLayout from '../../Layouts/FullscreenLayout.vue';
+import StudentExerciseGradeForm from './StudentExerciseGradeForm.vue';
+import { useArrayFilter } from '../../Utils/useArrayFilter.js';
+
+defineOptions({ layout: FullscreenLayout });
+
+const { course } = defineProps({
+  course: {
+    type: Object,
+    required: true,
+  },
+});
+
+const { exercises, students } = course;
+
+const [filteredExercises, applyExercisesFilter, currExercise] = useArrayFilter({
+  array: exercises,
+});
+
+const [filteredStudents, applyStudentsFilter, currStudent] = useArrayFilter({
+  array: students,
+});
+
+const getStudentName = (student) => {
+  return `${student.last_name}, ${student.first_name}`;
+};
+
+/**
+ * Combine student and exercise ids into a single unique id
+ * @param {Object} student
+ * @param {Object} exercise
+ */
+const getFormKey = (student, exercise) => `${student.id}-${exercise.id}`;
+
+/**
+ * Generates an object to store all combinations of student x exercise
+ */
+const getInitialMatrixValue = () => {
+  const initialMatrix = {};
+  students.forEach((student) => {
+    exercises.forEach((exercise) => {
+      const key = getFormKey(student, exercise);
+      initialMatrix[key] = {
+        student: student.id,
+        exercise: exercise.id,
+        comment: '',
+        grade_type: '',
+        grade_value: null,
+      };
+    });
+  });
+  return initialMatrix;
+};
+
+const form = useForm(getInitialMatrixValue());
+
+const handleSubmit = () => {
+  form
+    .transform((data) => ({
+      grades: Object.values(data),
+    }))
+    .post(`/courses/${course.id}/batchGrade`);
+};
+</script>
+
+<template>
+  <Head title="Full Screen - Grade Helper" />
+  <div class="d-flex flex-column flex-grow-1 vh-100">
+    <div class="flex-grow-1 overflow-auto" style="">
+      <table id="fullscreen-grades-table" class="table table-striped m-0">
+        <thead>
+          <tr>
+            <th scope="col"></th>
+            <th
+              v-for="exercise in filteredExercises"
+              :key="exercise.id"
+              scope="col"
+            >
+              {{ exercise.title }}
+
+              <button
+                class="btn btn-secondary btn-sm"
+                @click="applyExercisesFilter(exercise.id)"
+              >
+                {{ currExercise === exercise.id ? 'show all' : 'solo' }}
+              </button>
+            </th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr v-for="student in filteredStudents" :key="student.id">
+            <td class="sticky-top">
+              {{ getStudentName(student) }}
+              <button
+                class="btn btn-secondary btn-sm"
+                @click="applyStudentsFilter(student.id)"
+              >
+                {{ currStudent === student.id ? 'show all' : 'solo' }}
+              </button>
+            </td>
+            <td v-for="exercise in filteredExercises" :key="exercise.id">
```

Comment:
Acá creo que estaría bueno dejar un enlace al trabajo. De paso podríamos ver si está disponible o no usando el componente `CheckOnlineStatus`

```suggestion
            <td v-for="exercise in filteredExercises" :key="exercise.id">
              <span>
                <CheckOnlineStatus
                  :student-username="student.github_username"
                  :exercise-path="exercise.path"
                />
                <a
                  :href="`https://${student.github_username}.github.io/plataformas-moviles-entregas/${exercise.path}`"
                  >Link
                </a>
              </span>
```